src/librustc_metadata/schema.rs

```diff
@@ -175,13 +174,23 @@ pub struct CrateRoot<'tcx> {
     pub native_libraries: LazySeq<NativeLibrary>,
     pub foreign_modules: LazySeq<ForeignModule>,
     pub source_map: LazySeq<syntax_pos::SourceFile>,
-    pub def_path_table: Lazy<hir::map::definitions::DefPathTable>,
     pub impls: LazySeq<TraitImpls>,
     pub exported_symbols: LazySeq<(ExportedSymbol<'tcx>, SymbolExportLevel)>,
     pub interpret_alloc_index: LazySeq<u32>,
 
     pub entries_index: LazySeq<index::Index<'tcx>>,
 
+    /// The data for any proc macros declared by this crate
+    /// These are stored in the same order as the proc macros
+    /// are stored under their dylib symbol
+    /// This needs to come before 'def_path_table',
+    /// as we need use data from it when decoding `def_path_table.
+    /// This also needs to come at the very end of the 'Lazy/LazySeq' data,
+    /// as we need all of the other data in order to deserialize it
```

Comment:
The last two lines here don't make sense: a `Lazy`/`LazySeq` is just an offset into the file (hence "lazy"), so the order doesn't matter at all.

Reply:
I meant to refer to the [encoding order](https://github.com/rust-lang/rust/blob/60960a260f7b5c695fd0717311d72ce62dd4eb43/src/librustc_metadata/encoder.rs#L258-L259), which does matter. However, now that I lazily deserialize the proc macros, it doesn't matter.